.github/workflows/github_pages.yml

```diff
@@ -0,0 +1,61 @@
+name: Deploy to GitHub Pages
+on:
+  workflow_call:
+    inputs:
+      settings:
+        required: true
+        description: "The path to your Pelican settings file (`pelican`'s `--settings` option), for example: 'publishconf.py'"
+        type: string
+      requirements:
+        required: false
+        default: "pelican"
+        description: "The Python requirements to install, for example to enable markdown and typogrify use: 'pelican[markdown] typogrify'"
+        type: string
+      output-path:
+        required: false
+        default: "output/"
+        description: "Where to output the generated files (`pelican`'s `--output` option)"
+        type: string
+permissions:
+  contents: read
+  pages: write
+  id-token: write
+concurrency:
+  group: "pages"
+  cancel-in-progress: false
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: "3.11"
+      - name: Configure GitHub Pages
+        id: pages
+        uses: actions/configure-pages@v3
+      - name: Install requirements
+        run: pip install ${{ inputs.requirements }}
+      - name: Build Pelican site
+        run: |
+          pelican \
+            --settings "${{ inputs.settings }}" \
+            --extra-settings SITEURL='"${{ steps.pages.outputs.base_url }}"' \
+            --output "${{ inputs.output-path }}"
```

Comment:
For `--output` instead so that any `OUTPUT_PATH` in the user's Pelican settings is ignored. This is necessary because the upload-artifact step below needs to know what the output path is. The user can still customize the output path by passing an `output-path` input to the workflow